idseq_dag/steps/run_srst2.py

```diff
@@ -69,6 +72,76 @@ def get_common_params(self):
         return ['--min_coverage', min_cov,'--threads', n_threads,
                 '--output',  os.path.join(self.output_dir_local, 'output'), '--log', '--gene_db', db_file_path]
 
+    def generate_mapped_reads_table(self):
+        """Use bedtools to generate a table of mapped reads for each genome in the ARG ANNOT database.
+            If a new resistance gene db is used, the .bed file will need to be updated manually."""
+        bed_file_path = fetch_from_s3(self.additional_files["resist_genome_bed"], self.output_dir_local, allow_s3mi=False)
+        bedtools_params = ['bedtools', 'coverage', '-b', self.output_files_local()[5], '-a', bed_file_path, '>>', os.path.join(self.output_dir_local, 'matched_reads.tsv')]
+        command.execute(" ".join(bedtools_params))
+
+    def get_total_reads(self, is_zipped, is_fasta):
+        """Gets the total number of reads in the sample by counting them directly from the
+            fastq or fasta files."""
+        input_filenames = self.input_files_local[0]
+        if is_zipped:
+            gunzip_params = ['gunzip', '-k']
+            gunzip_params.extend(input_filenames)
+            command.execute(" ".join(gunzip_params))
+            input_filenames = map(lambda name: name[:len(name)-3], input_filenames)
+        if is_fasta:
+            grep_params = ['grep', '-c', '"^>"'] # fastas start reads with "^>"
+            grep_params.extend(input_filenames) 
+            grep_output = command.execute_with_output(" ".join(grep_params))
+            output_lines = [line for line in grep_output.split("\n") if line != '']
+            if ":" in output_lines[0]: # for paired fastas
+                read_counts = map(lambda line: int(line.split(":")[1]), output_lines)
+                return reduce(lambda x, y: x + y, list(read_counts))
+            else:
+                return int(output_lines[0])
+        else:
+            wc_params = ['wc', '-l']
+            wc_params.extend(input_filenames)
+            wc_output = command.execute_with_output(" ".join(wc_params))
+            # take the set of characters from the last line, which is the total number of lines
+            # for paired reads or the only line for unpaired reads
+            wc_lines = [line for line in wc_output.split("\n") if line != '']
+            wc_target_line = [line for line in wc_lines[-1].split(" ") if line != '']
+            total_line_count = ''.join(filter(lambda x: x.isdigit(), str(wc_target_line[0])))
```

Comment:
Is there a chance that some of the characters won't be a number? Curious why you do the filter.